common.mk

```diff
@@ -34,7 +34,18 @@ LIBRARIES+=$(wildcard $(FWDIR)/*.a)
 EXCLUDE_COLD_LIBRARIES+=$(FWDIR)/libc.a $(FWDIR)/libm.a
 COLD_LIBRARIES=$(filter-out $(EXCLUDE_COLD_LIBRARIES), $(LIBRARIES))
 wlprefix=-Wl,$(subst $(SPACE),$(COMMA),$1)
+
+ifneq (,$(wildcard ./firmware/liblvgl.a))
+    LVGL_LIB_FLAGS=--whole-archive ./firmware/liblvgl.a --no-whole-archive
+    FILTERED_LIBRARIES=$(filter-out ./firmware/liblvgl.a, $(LIBRARIES))
+else
+    LVGL_LIB_FLAGS=
+    FILTERED_LIBRARIES=$(LIBRARIES)
+endif
+
 LNK_FLAGS=--gc-sections --start-group $(strip $(LIBRARIES)) -lgcc -lstdc++ --end-group -T$(FWDIR)/v5-common.ld --no-warn-rwx-segments
+LNK_FLAGS_MONOLITH=--gc-sections --start-group $(strip $(LVGL_LIB_FLAGS) $(FILTERED_LIBRARIES)) -lgcc -lstdc++ --end-group -T$(FWDIR)/v5-common.ld --no-warn-rwx-segments
```

Comment:
Can we figure out a better way to do this than duplication? Also should `LNK_FLAGS` be `LNK_FLAGS_HOTCOLD` or something?